DEVDOCS-5710 [add]: Metafields, add metafields batch endpoints

bc-andreadao opened the pull request:
# [DEVDOCS-5710]

## What changed?
- Add endpoints of batch management of metafields

## Release notes draft
* The newly-released Metafield endpoints are now available to use. Now, you’ll be able to manage metafields in batches for all types of metafield, including those for product, product variants, and more.

## Related PR
- [PR 281](https://github.com/bigcommerce/api-specs-ssot/pull/281) for batch metafield endpoint for Locations API
- [PR 790](https://github.com/bigcommerce/developer-center/pull/790) for redirects

[DEVDOCS-5710]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-5710?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

slsriehl commented on reference/carts.v3.yml:
```diff
@@ -965,6 +965,142 @@ paths:
+                        type: string
+                        example: "42"
+                        description: |
+                          The id for the cart with which the metafield is associated.
```
```suggestion
                          The ID for the cart with which the metafield is associated.
```

capitalize throughout.

slsriehl commented on reference/carts.v3.yml:
```diff
@@ -965,6 +965,142 @@ paths:
+      tags:
+        - Metafields
+      description: Create multiple metafields.
+      operationId: updateCartsMetafields
```
```suggestion
      operationId: updateMetafieldsCarts
```

slsriehl commented on reference/carts.v3.yml:
```diff
@@ -965,6 +965,142 @@ paths:
+      tags:
+        - Metafields
+      description: Delete all cart metafields.
+      operationId: deleteCartMetafields
```
```suggestion
      operationId: deleteMetafieldsCarts
```

slsriehl commented on reference/carts.v3.yml:
```diff
@@ -3020,6 +3156,10 @@ components:
               example: '2022-06-16T18:39:00+00:00'
+            owner_client_id:
+              type: string
+              description: Client ID for the metafield's creator.
```
```suggestion
              description: Client ID for the metafieldʼs creator.
```

update throughout.

slsriehl commented on reference/carts.v3.yml:
```diff
@@ -3163,6 +3297,135 @@ components:
+            type: integer
+            description: |
+              The unique identifier for the metafield.
+          example: [
```
convert to YAML

slsriehl commented on reference/carts.v3.yml:
```diff
@@ -3163,6 +3297,135 @@ components:
+            type: integer
+            description: |
+              The unique identifier for the metafield.
+          example: [
```
convert everything to YAML throughout.

slsriehl commented on reference/carts.v3.yml:
```diff
@@ -3163,6 +3297,135 @@ components:
+      description: |
+        Error detail response payload for the BigCommerce API.
+      example:
+        {
```
yaml

slsriehl commented on reference/carts.v3.yml:
```diff
@@ -3163,6 +3297,135 @@ components:
+        }
+    WriteCollectionSuccessMeta:
+      type: object
+      description: 'Additional data about the response.'
```
no quotes needed. if quotes are being automatically added, you have stray single quote(s).

slsriehl commented on reference/carts.v3.yml:
```diff
@@ -3334,13 +3597,35 @@ components:
       name: namespace
       in: query
-      description: Filter based on a metafield's key.
+      description: Filter based on a metafield's namespaces.
```
```suggestion
      description: Filter based on a metafieldʼs namespaces.
```

slsriehl commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1587,6 +1587,142 @@ paths:
+      tags:
+        - Metafields
+      description: Get all brand metafields.
+      operationId: getAllBrandMetafields
```
```suggestion
      operationId: getMetafieldsBrands
```

slsriehl commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1587,6 +1587,142 @@ paths:
+      tags:
+        - Metafields
+      description: Create multiple metafields.
+      operationId: createBrandMetafields
```
```suggestion
      operationId: createMetafieldsBrands
```

slsriehl commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1587,6 +1587,142 @@ paths:
+                        type: string
+                        example: "42"
+                        description: |
+                          The id for the cart with which the metafield is associated.
```
```suggestion
                          The ID for the brand with which the metafield is associated.
```
check the cut and paste. capitalize ID throughout where not in code quotes.

slsriehl commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1587,6 +1587,142 @@ paths:
+      tags:
+        - Metafields
+      description: Create multiple metafields.
+      operationId: updateBrandMetafields
```
```suggestion
      operationId: updateMetafieldsBrands
```

slsriehl commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1587,6 +1587,142 @@ paths:
+      tags:
+        - Metafields
+      description: Delete all brand metafields.
+      operationId: deleteBrandMetafields
```
```suggestion
      operationId: deleteMetafieldsBrands
```

slsriehl commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1868,13 +2004,418 @@ components:
+            type: integer
+            description: |
+              The unique identifier for the metafield.
+          example: [
```
conver to YAML.

slsriehl commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1868,13 +2004,418 @@ components:
+      description: |
+        Error detail response payload for the BigCommerce API.
+      example:
+        {
```
YAML

slsriehl commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1908,6 +2449,76 @@ components:
+    MetafieldKeyInParam:
+      name: key:in
+      in: query
+      description: Filter based on comma separated metafieldʼs keys. Could be used with vanila 'key' query parameter.
```
```suggestion
      description: Filter based on comma separated metafieldʼs keys. Could be used with vanilla `key` query parameter.
```

slsriehl commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1908,6 +2449,76 @@ components:
+    MetafieldNamespaceInParam:
+      name: namespace:in
+      in: query
+      description: Filter based on comma separated metafieldʼs namespaces. Could be used with vanila 'namespace' query parameter.
```
```suggestion
      description: Filter based on comma separated metafieldʼs namespaces. Could be used with vanilla `namespace` query parameter.
```
correct vanilla and swap straight quotes for backticks throughout this series of params.

slsriehl commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1908,6 +2449,76 @@ components:
+          - desc
+  responses:
+    General207Status:
+      description: 'Multi-status. Multiple operations have taken place and the status for each operation can be viewed in the body of the response. Typically indicates that a partial failure has occured, such as when a `POST` or `PUT` request is successful, but saving the URL or inventory data has failed.'
```
```suggestion
      description: 'Multi-status. Multiple operations have taken place and the status for each operation can be viewed in the body of the response. Typically indicates that a partial failure has occured, such as when a `POST` or `PUT` request is successful, but saving the URL or inventory data has failed.'
```
straight quotes unnecessary. search for stray straight quotes throughout.

slsriehl commented on reference/catalog/categories_catalog.v3.yml:
```diff
@@ -1783,6 +1783,142 @@ paths:
+      tags:
+        - Metafields
+      description: Get all category metafields.
+      operationId: getAllCategoryMetafields
```
```suggestion
      operationId: getMetafieldsCategory
```

slsriehl commented on reference/catalog/categories_catalog.v3.yml:
```diff
@@ -1783,6 +1783,142 @@ paths:
+      tags:
+        - Metafields
+      description: Create multiple metafields.
+      operationId: createCategoryMetafields
```
```suggestion
      operationId: createMetafieldsCategories
```

slsriehl commented on reference/catalog/categories_catalog.v3.yml:
```diff
@@ -1783,6 +1783,142 @@ paths:
+      tags:
+        - Metafields
+      description: Create multiple metafields.
+      operationId: updateCategoryMetafields
```
```suggestion
      operationId: updateMetafieldsCategories
```

slsriehl commented on reference/catalog/categories_catalog.v3.yml:
```diff
@@ -1783,6 +1783,142 @@ paths:
+      tags:
+        - Metafields
+      description: Delete all category metafields.
+      operationId: deleteCategoryMetafields
```
```suggestion
      operationId: deleteMetafieldsCategories
```

slsriehl commented on reference/catalog/product-variants_catalog.v3.yml:
```diff
@@ -1731,6 +1731,142 @@ paths:
+      tags:
+        - Metafields
+      description: Get all variant metafields.
+      operationId: getVariantsMetafields
```
```suggestion
      operationId: getMetafieldsVariants
```

slsriehl commented on reference/catalog/product-variants_catalog.v3.yml:
```diff
@@ -1731,6 +1731,142 @@ paths:
+      tags:
+        - Metafields
+      description: Create multiple metafields.
+      operationId: updateVariantMetafields
```
```suggestion
      operationId: updateMetafieldsVariants
```

slsriehl commented on reference/catalog/product-variants_catalog.v3.yml:
```diff
@@ -1731,6 +1731,142 @@ paths:
+      tags:
+        - Metafields
+      description: Delete all variant metafields.
+      operationId: deleteVariantMetafields
```
```suggestion
      operationId: deleteMetafieldsVariants
```

slsriehl commented on reference/catalog/products_catalog.v3.yml:
```diff
@@ -6281,6 +6281,142 @@ paths:
+      tags:
+        - Metafields
+      description: Get all product metafields.
+      operationId: getProductsMetafields
```
```suggestion
      operationId: getMetafieldsProducts
```

slsriehl commented on reference/catalog/products_catalog.v3.yml:
```diff
@@ -6281,6 +6281,142 @@ paths:
+      tags:
+        - Metafields
+      description: Create multiple metafields.
+      operationId: createProductMetafields
```
```suggestion
      operationId: createMetafieldsProducts
```

slsriehl commented on reference/catalog/products_catalog.v3.yml:
```diff
@@ -6281,6 +6281,142 @@ paths:
+      tags:
+        - Metafields
+      description: Create multiple metafields.
+      operationId: updateProductMetafields
```
```suggestion
      operationId: updateMetafieldsProducts
```

slsriehl commented on reference/catalog/products_catalog.v3.yml:
```diff
@@ -6281,6 +6281,142 @@ paths:
+      tags:
+        - Metafields
+      description: Delete all product metafields.
+      operationId: deleteProductMetafields
```
```suggestion
      operationId: deleteMetafieldsProducts
```

slsriehl commented on reference/channels.v3.yml:
```diff
@@ -767,6 +767,122 @@ paths:
+      tags:
+        - Channel Metafields
+      description: Get all channel metafields.
+      operationId: getChannelsMetafields
```
```suggestion
      operationId: getMetafieldsChannels
```

slsriehl commented on reference/orders.v3.yml:
```diff
@@ -873,6 +873,142 @@ paths:
+      tags:
+        - Metafields
+      description: Get all order metafields.
+      operationId: getOrdersMetafields
```
```suggestion
      operationId: getMetafieldsOrders
```

slsriehl commented on reference/orders.v3.yml:
```diff
@@ -873,6 +873,142 @@ paths:
+      tags:
+        - Metafields
+      description: Create multiple metafields.
+      operationId: createOrderMetafields
```
```suggestion
      operationId: createMetafieldsOrders
```

slsriehl commented on reference/orders.v3.yml:
```diff
@@ -873,6 +873,142 @@ paths:
+      tags:
+        - Metafields
+      description: Create multiple metafields.
+      operationId: updateOrderMetafields
```
```suggestion
      operationId: updateMetafieldsOrders
```

slsriehl commented on reference/orders.v3.yml:
```diff
@@ -873,6 +873,142 @@ paths:
+      tags:
+        - Metafields
+      description: Delete all order metafields.
+      operationId: deleteOrderMetafields
```
```suggestion
      operationId: deleteMetafieldsOrders
```

bc-tayvs commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1588,6 +1588,142 @@ paths:
+                  - type: object
+                    properties:
+                      resource_id:
+                        type: string
```
Should be an int

bc-andreadao replied:
✔️

bc-tayvs commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1588,6 +1588,142 @@ paths:
+                        description: |
+                          The ID of metafield to update.
+                    required:
+                      - id
```
resource_id

bc-tayvs commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1588,6 +1588,142 @@ paths:
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MetaFieldCollectionDeleteResponseSuccess'
```
Look like 422 response. Should be without errors field

bc-tayvs replied:
errors should be empty list 🤔 
<img width="563" alt="image" src="https://github.com/bigcommerce/docs/assets/80041457/27a641a2-089d-482b-8d09-3092ca5fa58b">

bc-andreadao replied:
✔️

bc-tayvs commented on reference/catalog/categories_catalog.v3.yml:
```diff
@@ -1783,6 +1783,142 @@ paths:
+                  - type: object
+                    properties:
+                      resource_id:
+                        type: string
```
should be integer

bc-andreadao replied:
✔️

bc-tayvs commented on reference/catalog/categories_catalog.v3.yml:
```diff
@@ -1783,6 +1783,142 @@ paths:
+                  - type: object
+                    properties:
+                      id:
+                        type: string
```
integer

bc-andreadao replied:
✔️

bc-tayvs commented on reference/catalog/categories_catalog.v3.yml:
```diff
@@ -1783,6 +1783,142 @@ paths:
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MetaFieldCollectionResponse'
```
should have errors field. LIke this one

bc-tayvs replied:
https://github.com/bigcommerce/docs/pull/63/files#diff-c82ef9b50d12a202e1828e96fa92f381dffb60e5414f98590cbf7945d8330cc2R1726

bc-andreadao replied:
✔️

bc-tayvs commented on reference/catalog/categories_catalog.v3.yml:
```diff
@@ -1783,6 +1783,142 @@ paths:
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MetaFieldCollectionDeleteResponseSuccess'
```
should be with always empty errors field

bc-andreadao replied:
✔️

bc-tayvs commented on reference/catalog/product-variants_catalog.v3.yml:
```diff
@@ -1717,6 +1716,142 @@ paths:
+                  - type: object
+                    properties:
+                      resource_id:
+                        type: string
```
integer. All metafields except carts have resource_id int

bc-andreadao replied:
✔️

bc-tayvs commented on reference/catalog/product-variants_catalog.v3.yml:
```diff
@@ -1717,6 +1716,142 @@ paths:
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MetaFieldCollectionDeleteResponseSuccess'
```
Errors field should be empty. This statement true for all types of metafields. I think, it renders not empty because of example field or smth similar. Screenshot from swagger online editor
![image](https://github.com/bigcommerce/docs/assets/80041457/c54efbde-6272-4692-8159-b1455dbc97fa)

bc-andreadao replied:
✔️

bc-tayvs commented:
I also found some incorrect information in old code. THis parameters is not supported for sure

bc-tayvs commented:
and with some PUT endpoints as well
https://github.com/bigcommerce/docs/blob/23d4a9df96b6aa444c84e60cd6ec17833f86fc47/reference/catalog/brands_catalog.v3.yml#L1380-L1400

headers are nor required AFAIK and requestBody should be not metafield_Base but MetafieldBase_Put (all fields are optional)

bc-tayvs commented:
Tried without headers and it still works
<img width="1512" alt="image" src="https://github.com/bigcommerce/docs/assets/80041457/c0068729-0dde-4b52-bc6a-5b61a79f9de2">

bc-andreadao commented:
@bc-tayvs Thanks for the review. 

Based on your comments (and my testing), I changed the following for the responses:
- 200 responses for POST, PUT, and DELETE methods all have an empty `errors` array 
- 422 responses for the POST, PUT, and DELETE methods all have an empty `data` array. 

 I also changed the type for `resource_id` to integer. 
 
 For your comments on old code/endpoints: I created a new ticket ([DEVDOCS-5770](https://bigcommercecloud.atlassian.net/browse/DEVDOCS-5770)), so this PR can focus only on batch endpoints.
 
 Let me know if there's anything else

bc-tayvs commented:
@bc-andreadao

> 422 responses for the POST, PUT, and DELETE methods all have an empty data array.

422 is a partial success, so result could contain data as well as errors. Plz, make data field not empty for this case.
Thanks in advance.

bc-andreadao commented:
@bc-tayvs Thanks, just changed the 422 responses so data field isn't empty

bc-traciporter commented on reference/carts.v3.yml:
```diff
@@ -796,7 +762,7 @@ paths:
         '409':
           description: |
-            The `Metafield` conflicts with another `Metafield`. This can be the result of duplicate unique key combinations of the app's client id, namespace, key, resource_type, and resource_id.
+            The `Metafield` conflicts with another `Metafield`. This can be the result of duplicate unique key combinations of the appʼs client id, namespace, key, resource_type, and resource_id.
```
```suggestion
            The `Metafield` conflicts with another `Metafield`. This can result from duplicate unique key combinations of the appʼs client id, namespace, key, resource_type, and resource_id.
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/carts.v3.yml:
```diff
@@ -885,7 +849,7 @@ paths:
         '409':
           description: |
-            The `Metafield` conflicts with another `Metafield`. This can be the result of duplicate unique key combinations of the app's client id, namespace, key, resource_type, and resource_id.
+            The `Metafield` conflicts with another `Metafield`. This can be the result of duplicate unique key combinations of the appʼs client id, namespace, key, resource_type, and resource_id.
```
```suggestion
            The `Metafield` conflicts with another `Metafield`. This can result from duplicate unique key combinations of the appʼs client id, namespace, key, resource_type, and resource_id.
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/carts.v3.yml:
```diff
@@ -3330,17 +3583,39 @@ components:
+    MetafieldKeyInParam:
+      name: key:in
+      in: query
+      description: Filter based on comma separated metafieldʼs keys. Could be used with vanilla `key` query parameter.
```
```suggestion
      description: Filter based on comma-separated metafieldʼs keys. Could be used with vanilla `key` query parameter.
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/carts.v3.yml:
```diff
@@ -3330,17 +3583,39 @@ components:
+    MetafieldNamespaceInParam:
+      name: namespace:in
+      in: query
+      description: Filter based on comma separated metafieldʼs namespaces. Could be used with vanilla `namespace` query parameter.
```
```suggestion
      description: Filter based on comma-separated metafieldʼs namespaces. Could be used with vanilla `namespace` query parameter.
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1869,13 +2006,444 @@ components:
+            | `app_only` | Private to the app that owns the field. |
+            | `read` | Visible to other API consumers. |
+            | `write` | Open for reading and writing by other API consumers. |
+            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
```
```suggestion
            | `read_and_sf_access` | Visible to other API consumers, including on the storefront. |
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1869,13 +2006,444 @@ components:
+            | `read` | Visible to other API consumers. |
+            | `write` | Open for reading and writing by other API consumers. |
+            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
+            | `write_and_sf_access` | Open for reading and writing by other API consumers, including on storefront. |
```
```suggestion
            | `write_and_sf_access` | Open for reading and writing by other API consumers, including on the storefront. |
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1869,13 +2006,444 @@ components:
+            | `app_only` | Private to the app that owns the field. |
+            | `read` | Visible to other API consumers. |
+            | `write` | Open for reading and writing by other API consumers. |
+            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
```
```suggestion
            | `read_and_sf_access` | Visible to other API consumers, including on the storefront. |
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1869,13 +2006,444 @@ components:
+            | `read` | Visible to other API consumers. |
+            | `write` | Open for reading and writing by other API consumers. |
+            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
+            | `write_and_sf_access` | Open for reading and writing by other API consumers, including on storefront. |
```
```suggestion
            | `write_and_sf_access` | Open for reading and writing by other API consumers, including on the storefront. |
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1869,13 +2006,444 @@ components:
+            | `app_only` | Private to the app that owns the field. |
+            | `read` | Visible to other API consumers. |
+            | `write` | Open for reading and writing by other API consumers. |
+            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
```
```suggestion
            | `read_and_sf_access` | Visible to other API consumers, including on the storefront. |
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1869,13 +2006,444 @@ components:
+            | `read` | Visible to other API consumers. |
+            | `write` | Open for reading and writing by other API consumers. |
+            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
+            | `write_and_sf_access` | Open for reading and writing by other API consumers, including on storefront. |
```
```suggestion
            | `write_and_sf_access` | Open for reading and writing by other API consumers, including on the storefront. |
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/brands_catalog.v3.yml:
```diff
@@ -1909,6 +2477,76 @@ components:
+    MetafieldKeyInParam:
+      name: key:in
+      in: query
+      description: Filter based on comma separated metafieldʼs keys. Could be used with vanilla `key` query parameter.
```
```suggestion
      description: Filter based on comma-separated metafieldʼs keys. Could be used with vanilla `key` query parameter.
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/categories_catalog.v3.yml:
```diff
@@ -2301,9 +2437,442 @@ components:
+            | `app_only` | Private to the app that owns the field. |
+            | `read` | Visible to other API consumers. |
+            | `write` | Open for reading and writing by other API consumers. |
+            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
```
```suggestion
            | `read_and_sf_access` | Visible to other API consumers, including on the storefront. |
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/categories_catalog.v3.yml:
```diff
@@ -2301,9 +2437,442 @@ components:
+            | `read` | Visible to other API consumers. |
+            | `write` | Open for reading and writing by other API consumers. |
+            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
+            | `write_and_sf_access` | Open for reading and writing by other API consumers, including on storefront. |
```
```suggestion
            | `write_and_sf_access` | Open for reading and writing by other API consumers, including on the storefront. |
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/categories_catalog.v3.yml:
```diff
@@ -2301,9 +2437,442 @@ components:
+            | `app_only` | Private to the app that owns the field. |
+            | `read` | Visible to other API consumers. |
+            | `write` | Open for reading and writing by other API consumers. |
+            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
```
```suggestion
            | `read_and_sf_access` | Visible to other API consumers, including on the storefront. |
```

bc-traciporter commented on reference/catalog/categories_catalog.v3.yml:
```diff
@@ -2301,9 +2437,442 @@ components:
+            | `read` | Visible to other API consumers. |
+            | `write` | Open for reading and writing by other API consumers. |
+            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
+            | `write_and_sf_access` | Open for reading and writing by other API consumers, including on storefront. |
```
```suggestion
            | `write_and_sf_access` | Open for reading and writing by other API consumers, including on the storefront. |
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/categories_catalog.v3.yml:
```diff
@@ -2301,9 +2437,442 @@ components:
+            | `app_only` | Private to the app that owns the field. |
+            | `read` | Visible to other API consumers. |
+            | `write` | Open for reading and writing by other API consumers. |
+            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
```
```suggestion
            | `read_and_sf_access` | Visible to other API consumers, including on the storefront. |
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/categories_catalog.v3.yml:
```diff
@@ -2301,9 +2437,442 @@ components:
+            | `read` | Visible to other API consumers. |
+            | `write` | Open for reading and writing by other API consumers. |
+            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
+            | `write_and_sf_access` | Open for reading and writing by other API consumers, including on storefront. |
```
```suggestion
            | `write_and_sf_access` | Open for reading and writing by other API consumers, including on the storefront. |
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/categories_catalog.v3.yml:
```diff
@@ -2344,6 +2913,69 @@ components:
+    MetafieldKeyInParam:
+      name: key:in
+      in: query
+      description: Filter based on comma separated metafieldʼs keys. Could be used with vanilla `key` query parameter.
```
```suggestion
      description: Filter based on comma-separated metafieldʼs keys. Could be used with vanilla `key` query parameter.
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/categories_catalog.v3.yml:
```diff
@@ -2344,6 +2913,69 @@ components:
+    MetafieldNamespaceInParam:
+      name: namespace:in
+      in: query
+      description: Filter based on comma separated metafieldʼs namespaces. Could be used with vanilla `namespace` query parameter.
```
```suggestion
      description: Filter based on comma-separated metafieldʼs namespaces. Could be used with vanilla `namespace` query parameter.
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/product-variants_catalog.v3.yml:
```diff
@@ -2355,6 +2491,439 @@ components:
+            | `app_only` | Private to the app that owns the field. |
+            | `read` | Visible to other API consumers. |
+            | `write` | Open for reading and writing by other API consumers. |
+            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
```
```suggestion
            | `read_and_sf_access` | Visible to other API consumers, including on the storefront. |
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/product-variants_catalog.v3.yml:
```diff
@@ -2355,6 +2491,439 @@ components:
+            | `read` | Visible to other API consumers. |
+            | `write` | Open for reading and writing by other API consumers. |
+            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
+            | `write_and_sf_access` | Open for reading and writing by other API consumers, including on storefront. |
```
```suggestion
            | `write_and_sf_access` | Open for reading and writing by other API consumers, including on the storefront. |
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/product-variants_catalog.v3.yml:
```diff
@@ -2355,6 +2491,439 @@ components:
+            | `app_only` | Private to the app that owns the field. |
+            | `read` | Visible to other API consumers. |
+            | `write` | Open for reading and writing by other API consumers. |
+            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
```
```suggestion
            | `read_and_sf_access` | Visible to other API consumers, including on the storefront. |
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/product-variants_catalog.v3.yml:
```diff
@@ -2355,6 +2491,439 @@ components:
+            | `read` | Visible to other API consumers. |
+            | `write` | Open for reading and writing by other API consumers. |
+            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
+            | `write_and_sf_access` | Open for reading and writing by other API consumers, including on storefront. |
```
```suggestion
            | `write_and_sf_access` | Open for reading and writing by other API consumers, including on the storefront. |
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/product-variants_catalog.v3.yml:
```diff
@@ -2355,6 +2491,439 @@ components:
+            | `app_only` | Private to the app that owns the field. |
+            | `read` | Visible to other API consumers. |
+            | `write` | Open for reading and writing by other API consumers. |
+            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
```
```suggestion
            | `read_and_sf_access` | Visible to other API consumers, including on the storefront. |
```

bc-andreadao replied:
✔️

bc-traciporter commented on reference/catalog/product-variants_catalog.v3.yml:
```diff
@@ -2355,6 +2491,439 @@ components:
+            | `read` | Visible to other API consumers. |
+            | `write` | Open for reading and writing by other API consumers. |
+            | `read_and_sf_access` | Visible to other API consumers, including on storefront. |
+            | `write_and_sf_access` | Open for reading and writing by other API consumers, including on storefront. |
```
```suggestion
            | `write_and_sf_access` | Open for reading and writing by other API consumers, including on the storefront. |
```

bc-andreadao commented:
@bc-tayvs I created a new [PR 141](https://github.com/bigcommerce/docs/pull/141) to address your comments, if you can review
